File: frontend/src/utils/correoInstitucional.ts
import { apiClient } from "@/lib/api";

export const DOMINIOS_INSTITUCIONALES_DEFAULT = [
  "gmail.com",
  "outlook.com",
  "hotmail.com",
  "live.com",
];

let dominiosCache: string[] | null = null;

export function dominiosInstitucionales(): string[] {
  return dominiosCache ?? DOMINIOS_INSTITUCIONALES_DEFAULT;
}

export function esCorreoInstitucional(
  email: string,
  dominios: string[] = dominiosInstitucionales(),
): boolean {
  const dominio = email.trim().toLowerCase().split("@")[1];
  if (!dominio) return false;
  return dominios.some((item) => item.toLowerCase() === dominio);
}

export function mensajeCorreoInstitucional(
  dominios: string[] = dominiosInstitucionales(),
): string {
  const lista = dominios.map((item) => `@${item}`).join(", ");
  return `Debe usar un correo de Gmail o Outlook (${lista})`;
}

export async function cargarDominiosInstitucionales(): Promise<string[]> {
  try {
    const { data } = await apiClient.get("/auth/politica-acceso");
    if (Array.isArray(data?.dominios_institucionales) && data.dominios_institucionales.length) {
      dominiosCache = data.dominios_institucionales;
    }
  } catch {
    // Conserva los dominios por defecto si el backend no responde
  }
  return dominiosInstitucionales();
}

File: backend/app/utils/correo_institucional.py
"""Validación de correos institucionales para usuarios nuevos."""
from typing import List

from ..config import settings


def dominios_institucionales() -> List[str]:
    """Dominios permitidos para cuentas nuevas (OTP)."""
    return [
        dominio.strip().lower().lstrip("@")
        for dominio in settings.CORREOS_INSTITUCIONALES.split(",")
        if dominio.strip()
    ]


def es_correo_institucional(email: str) -> bool:
    if not email or "@" not in str(email):
        return False
    dominio = str(email).rsplit("@", 1)[-1].strip().lower()
    return dominio in set(dominios_institucionales())


def mensaje_correo_institucional() -> str:
    lista = ", ".join(f"@{dominio}" for dominio in dominios_institucionales())
    return (
        "Debe usar un correo de Gmail o Outlook "
        f"({lista})."
    )
